DeepLinkSDK/Router/DPLDeepLinkRouter.m

```diff
@@ -48,16 +52,31 @@ - (void)registerHandlerClass:(Class <DPLRouteHandler>)handlerClass forRoute:(NSS
     if (handlerClass && [route length]) {
         [self.routes addObject:route];
         [self.blocksByRoute removeObjectForKey:route];
+        [self.protocolsByRoute removeObjectForKey:route];
+        
         self.classesByRoute[route] = handlerClass;
     }
 }
 
+- (void)registerHandler:(id <DPLRouteHandlerProtocol>)handler forRoute:(NSString *)route {
+    
+    if (handler && [route length]) {
+        [self.routes addObject:route];
+        [self.blocksByRoute removeObjectForKey:route];
+        [self.classesByRoute removeObjectForKey:route];
+        
+        self.protocolsByRoute[route] = handler;
```

Comment:
I don't think we need to collect protocols separately. The way I see it, we could instead register classes that conform to the `DPLRouteHandlerProtocol` and have one class available `DPLRouteHandler` that already conforms to it with the default behavior.